### PR TITLE
Use -EBADF for fd initialization

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: coccinelle/safe_close.cocci [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
 - close(fd);
-- fd = -1;
+- fd = -EBADF;
 + fd = safe_close(fd);
 @@
 expression fd;
 @@
 - close_nointr(fd);
-- fd = -1;
+- fd = -EBADF;
```
Comment (poettering) [inline coccinelle/safe_close.cocci:12]: i guess one could write an expression here that allows either -1 or -EBADF in the match expression. but i too would be too lazy for that ;-)

File: src/basic/terminal-util.c [hunk 2]
```diff
@@ -1130,7 +1130,7 @@ static int ptsname_namespace(int pty, char **ret) {
 
 int openpt_allocate_in_namespace(pid_t pid, int flags, char **ret_slave) {
         _cleanup_close_ int pidnsfd = -EBADF, mntnsfd = -EBADF, usernsfd = -EBADF, rootfd = -EBADF, fd = -EBADF;
-        _cleanup_close_pair_ int pair[2] = { -1, -1 };
+        _cleanup_close_pair_ int pair[2] = { -EBADF, -EBADF };
```
Comment (poettering) [inline src/basic/terminal-util.c:1133]: might be worth adding PIPE_EBADF or so as macro for this? dunno

**Comment by poettering** [comment]
the codeql CI failure looks suspicious?

**Comment by keszybz** [comment]
> the codeql CI failure looks suspicious?

I think the code is correct. The test does `assert` that the return value is positive, so we expect the variable to be initialized.

**Pull request merged**